### PR TITLE
Added support for `dbutils.secrets`

### PR DESCRIPTION
**Pull request by tanmay-db**
## Changes

Add support for dbutils.secrets. New open api spec has been created to support secrets/get

TODO: 
1. Update https://github.com/databricks/databricks-sdk-java/blob/main/.codegen/_openapi_sha once the PR with spec changes is merged and we have the sha
2. Uncomment secrets/get section in integration test once it is enabled for test infra

## Tests

Unit and Integration tests

### CODE REVIEW

File: databricks-sdk-java/src/test/java/com/databricks/sdk/integration/SecretsIT.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+//    > }
+//    < 200 OK
+//    < { }
+//    16:06 [DEBUG] > GET /api/2.0/secrets/get?key=testKey-826&scope=testScope-826
+//    < 400 Bad Request
+//    < {
+//    <   "error_code" : "BAD_REQUEST",
+//    <   "message" : "This operation is not allowed outside databricks notebooks."
+//    < }
+//    =================   Uncomment once enabled ==================
```
Comment (tanmay-db) [inline databricks-sdk-java/src/test/java/com/databricks/sdk/integration/SecretsIT.java:35]: cc: @mgyucht on this for visibility

File: databricks-sdk-java/src/test/java/com/databricks/sdk/mixin/SecretsExtTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public class SecretsExtTest {
+  @Mock SecretsService mockedSecretsService;
+
+  @BeforeEach
+  void setup() {
+    MockitoAnnotations.openMocks(this);
```
Comment (mgyucht) [inline databricks-sdk-java/src/test/java/com/databricks/sdk/mixin/SecretsExtTest.java:21]: I believe this returns an `Autocloseable` that needs to be closed after each test, if I'm not mistaken.

**Review by mgyucht — COMMENTED** [review]
Couple comments, otherwise looking great!

File: databricks-sdk-java/src/main/java/com/databricks/sdk/service/workspace/GetSecretResponse.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
+// Code generated from OpenAPI specs by Databricks SDK Generator. DO NOT EDIT.
```
[the remark this answers is not in the document]
Comment (tanmay-db) [reply databricks-sdk-java/src/main/java/com/databricks/sdk/service/workspace/GetSecretResponse.java:1]: Yes, I forgot to add `.gitattributes`.

**Pull request merged**